File: EncapsulationAndPolymorphism/01.Shapes/Circle.cs
using System;

namespace Shapes
{
    class Circle : IShape
    {
        private double radius;

        public Circle(double radius)
        {
            this.Radius = radius;
        }

        public double Radius
        {
            get { return this.radius; }
            set
            {
                if (value < 0)
                {
                    throw new ArgumentOutOfRangeException("Write positive number");
                }
                this.radius = value;
            }
        }

        public double CalculateArea()
        {
            return Math.PI * Math.Pow(this.Radius, 2);
        }

        public double CalculatePerimeter()
        {
            return 2 * Math.PI * this.Radius;
        }
    }
}

File: OOPinJava/01.Geometry/src/shapes/planeShape/Circle.java
package shapes.planeShape;

import java.util.Arrays;

public class Circle extends PlaneShape {

    private double radius;

    public Circle(double[] vertex, double radius) throws Exception {
        super(vertex);
        this.setRadius(radius);
    }

    public double getRadius() {
        return this.radius;
    }

    public void setRadius(double radius) throws Exception {
        if (radius <= 0) {
            throw new Exception("The radius should be positive number.");
        }
        this.radius = radius;
    }

    @Override
    public double getArea() {
        return Math.PI * this.getRadius() * this.getRadius();
    }

    @Override
    public double getPerimeter() {
        return 2 * Math.PI * this.getRadius();
    }
    
    @Override
    public String toString() {
        return "Circle:"
                + "\n\tVerex: " + Arrays.toString(this.vertex)
                + "\n\tRadius: " + this.getRadius()
                + "\n\tPerimeter: " + this.getPerimeter()
                + "\n\tArea: " + this.getArea();
    }
}
